crates/router/src/connector/worldline.rs

```diff
@@ -0,0 +1,663 @@
+mod transformers;
+
+use std::fmt::Debug;
+
+use base64::Engine;
+use bytes::Bytes;
+use error_stack::{IntoReport, ResultExt};
+use ring::hmac;
+use time::{format_description, OffsetDateTime};
+use transformers as worldline;
+
+use crate::{
+    configs::settings::Connectors,
+    consts,
+    core::errors::{self, CustomResult},
+    headers, logger,
+    services::{self, ConnectorIntegration},
+    types::{
+        self,
+        api::{self, ConnectorCommon},
+        ErrorResponse, Response,
+    },
+    utils::{self, BytesExt, OptionExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Worldline;
+
+impl Worldline {
+    pub fn generate_authorization_token(
+        &self,
+        auth: worldline::AuthType,
+        http_method: &str,
+        content_type: &str,
+        date: &str,
+        endpoint: &str,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let signature_data: String = http_method.trim().to_owned()
+            + "\n"
+            + content_type.trim()
+            + "\n"
+            + date.trim()
+            + "\n/"
+            + endpoint.trim()
+            + "\n";
+        let worldline::AuthType {
+            api_key,
+            api_secret,
+            merchant_account_id: _,
+        } = auth;
+        let key = hmac::Key::new(hmac::HMAC_SHA256, api_secret.as_bytes());
+        let signed_data = base64::engine::general_purpose::STANDARD
+            .encode(hmac::sign(&key, signature_data.as_bytes()));
+
+        Ok(format!("GCS v1HMAC:{api_key}:{signed_data}"))
+    }
+
+    pub fn get_current_date_time(&self) -> CustomResult<String, errors::ConnectorError> {
```

Comment:
Do you need `self` at all? If not, remove it.

Reply:
Done.